### DISCUSSION TITLE
Making scatter timeline graph using chart.js

### DISCUSSION THREAD

**Discussion by gyomin0315** [community/community#67686] [Programming Help]
### Select Topic Area

Question

### Body

I'm trying to make scatter-plot timeline using chart.js. 
I organized dates to be in X-axes, but instead of dates, some numbers appeared in X-axes. 
How can I show dates in X-axes using scatter-plot in chart.js ? 
Here is html and js code. 
---------------------------------------------html--------------------------------------------------------------------
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Basic Graph with vis.js</title>
    <script src="https://kit.fontawesome.com/2751a91418.js" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>
    
    <link rel="stylesheet" href="C:\Users\82107\vis-4.21.0\vis-4.21.0\dist\vis-network.min.css">
    <link rel="stylesheet" href="C:\Users\82107\vis-4.21.0\vis-4.21.0\dist\vis.min.css">
</head>
<body>
    <script src="C:\Users\82107\vis-4.21.0\vis-4.21.0\dist\vis-network.min.js"></script>
    <script src="C:\Users\82107\vis-4.21.0\vis-4.21.0\dist\vis.min.js"></script>
    <div id="graph-container" style="width: 1200px; height: 1000px;"></div>
    <div id="graph-container2" style="width: 1200px; height: 1000px;"></div>
    <canvas id="scatter-plot"></canvas>
    <script src="app.js"></script>
    <style>
        #graph-container {
            background-color: #F0F8FF;
        }
        #graph-container2 {
            background-color: #F0F8FF;
        }
        #scatter-plot {
            max-width: 500px; 
            max-height: 300px; 
            overflow: auto; 
            background-color: #F0F8FF;
        }
        
    </style>
</body>
</html>
----------------------------------------------------------------------------------------------------------------------------------------------

-------------------------------------------------------------javascript-----------------------------------------------------------------------
var dates = ['2023-09-10', '2023-09-11', '2023-09-12'];
var cases = [5, 10, 15];

// 데이터셋을 생성합니다.
var dataset = {
    labels: dates.map(date => new Date(date)), // 문자열을 Date 객체로 변환합니다.
    datasets: [{
        label: 'Case',
        data: cases,
        borderColor: 'rgba(75, 192, 192, 1)', // 선의 색상
        backgroundColor: 'rgba(75, 192, 192, 0.5)', // 점의 배경색
        pointRadius: 5, // 점의 반지름
        pointHoverRadius: 7, // 마우스 오버 시 점의 반지름
        fill: false // 점과 점 사이를 선으로 연결하지 않습니다.
    }]
};

// 그래프 옵션을 설정합니다.
var options = {
    responsive: false,
    scales: {
        x: [{
            type: 'time', // x 축의 데이터 타입을 시간으로 설정합니다.
            time: {
                unit: 'day', // x 축에 표시할 시간 단위를 '일'로 설정합니다.
                displayFormats: {
                    day: 'YYYY-MM-DD' // x 축에 날짜를 표시할 서식을 지정합니다.
                }
            },
            title: {
                display: true,
                text: '날짜' // x 축 제목
            }
        }],
        y: {
            title: {
                display: true,
                text: 'Case' // y 축 제목
            }
        }
    }
};

// 캔버스 요소를 선택하고 점 그래프를 그립니다.
var ctx = document.getElementById('scatter-plot').getContext('2d');
var scatterChart = new Chart(ctx, {
    type: 'scatter', // 점 그래프 유형을 'scatter'로 설정합니다.
    data: dataset,
    options: options
});
----------------------------------------------------------------------------------------------------------------------------------------------
ps. graph-container and graph-container2 works fine. All that matters is scatter-plot.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @gyomin0315 ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
